### ISSUE TITLE
"Permission denied" when running uv in docker + Debian bookworm

### ISSUE THREAD

**Issue by torarvid** [astral-sh/uv#9962]
Hello and thanks for the wonderful `uv`. 🙌🏻

When running a docker container based on the below Dockerfile, I get this error:

```sh
error: failed to create directory `/nonexistent/.cache/uv`: Permission denied (os error 13)
```

This issue is likely not an issue with `uv` itself, but mentioning it in case others are having the same issue as me.

## Steps to reproduce

1. Make a folder `myproject` and cd into it
2. Run `uv init` (to create an empty-ish pyproject.toml and an example hello.py)
3. Make a Dockerfile (see below)
4. Run `docker build . -t myproject && docker run myproject`

Dockerfile contents:
```dockerfile
FROM python:3.12.7-slim-bookworm

RUN addgroup --system appuser && adduser --system --group appuser
RUN mkdir /app && chown appuser:appuser /app
WORKDIR /app
COPY --chown=appuser:appuser . /app/

COPY --from=ghcr.io/astral-sh/uv:0.5.9 /uv /bin/uv
USER appuser
RUN uv sync

CMD ["uv", "run", "hello.py"]
```

The reason I believe it's not necessarily an issue with `uv` is that if you replace `bookworm` with `bullseye` (the previous version of Debian), it works fine.

**Comment by torarvid** [comment]
@konstin It is the real path; the line was copy-pasted verbatim from my terminal.

**Comment by konstin** [comment]
Creating the user sets its home to the `/nonexistent` directory, but doesn't create that directory:

```
$ addgroup --system appuser && adduser --system --group appuser
Adding group `appuser' (GID 101) ...
Done.
Adding system user `appuser' (UID 100) ...
Adding new user `appuser' (UID 100) with group `appuser' ...
Not creating `/nonexistent'.
```

```
$ cat /etc/passwd | grep appuser
appuser:x:100:101::/nonexistent:/usr/sbin/nologin
```

uv needs a user cache directory (following the XDG rules), it doesn't with a system user with no home.

**Comment by torarvid** [comment]
@konstin heh, you beat me to it (I was testing and saw the same as you)

So for some reason it looks like Debian bullseye creates a HOME directory, but Debian bookworm no longer does (??)

I have no idea why that would be, but it seems to perfectly explain why it happens.

I changed the line in the Dockerfile to fix it:

```diff
-RUN addgroup --system appuser && adduser --system --group appuser
+RUN addgroup --system appuser && adduser --system --group --home /home/appuser appuser
```

Ok, I'm gonna consider this "problem solved", and (obviously) not `uv`s fault. Sorry if I created noise — hope it somehow helps somebody else out. Have a nice day!
